dashboard/client/src/pages/serve/hook/useServeApplications.ts

```diff
@@ -118,3 +123,42 @@ export const useServeApplicationDetails = (
     allDeployments: deployments,
   };
 };
+
+export const useServeReplicaDetails = (
+  applicationName: string | undefined,
+  deploymentName: string | undefined,
+  replicaId: string | undefined,
+) => {
+  // TODO(aguo): Use a fetch by replicaId endpoint?
+  const { data, error } = useSWR(
+    "useServeReplicaDetails",
+    async () => {
+      const rsp = await getServeApplications();
+
+      if (rsp) {
+        return rsp.data;
+      }
+    },
+    { refreshInterval: API_REFRESH_INTERVAL_MS },
+  );
+
+  const application = applicationName
+    ? data?.applications?.[applicationName !== "-" ? applicationName : ""]
```

Comment:
could the applicationName be "-"? it's sourced from `useParams` and i think "-" elements are non-clickable

Reply:
Yes, that's an edge case if a user names their application "-". But it's a  lot more common for an app to be nameless than to be named "-" so we should handle the "" case seamlessly.